### PR TITLE
Execute a pipeline

### PR DESCRIPTION
**Pull request by mikerowehl**
Has an option to poll for completion.

### CODE REVIEW

File: actions.go [hunk 1]
```diff
@@ -15,6 +15,52 @@ import (
 )
 
 // PipelineSaveAction creates the ActionFunc for saving pipeline configurations.
+func PipelineExecAction(clientConfig spinnaker.ClientConfig) cli.ActionFunc {
```
Comment (robzienert) [inline actions.go]: I merged a doc PR just before yours, so this function needs a comment (and the one that's currently right above it needs to move to the function below it).

File: actions.go [hunk 2]
```diff
@@ -15,6 +15,52 @@ import (
...
+		pipelineName := cc.Args().Get(1)
+		monitor := cc.Bool("monitor")
+		numRetries := cc.Int("retry")
+
+		logrus.WithFields(logrus.Fields{
+			"app":      appName,
+			"pipeline": pipelineName,
+			"monitor": 	monitor,
+			"retries": 	numRetries,
+		}).Info("Executing Pipeline...")
```
Comment (robzienert) [inline actions.go]: Run `go fmt`

File: actions.go [hunk 3]
```diff
@@ -15,6 +15,52 @@ import (
...
+		client, err := clientFromContext(cc, clientConfig)
+		if err != nil {
+			return errors.Wrapf(err, "creating spinnaker client")
+		}
+
+		resp, err := client.ExecPipeline(appName, pipelineName)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't execute pipeline")
+		}
+		fmt.Printf("Ref task id: %s\n", resp.Ref)
```
Comment (robzienert) [inline actions.go]: And switch all `fmt.Printf` to using logrus.

File: spinnaker/http.go [hunk 4]
```diff
@@ -100,7 +100,7 @@ func (c *client) postJSON(url string, body interface{}) (resp *http.Response, re
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "marshaling body to json")
 	}
-
+	logrus.WithField("url", url).Debug("http post")
```
Comment (robzienert) [inline spinnaker/http.go]: Needed?

**Comment by mikerowehl** [comment]
Ugh, sorry Rob. I thought that branch was clean, but apparently not. Fixed up all that stuff and rebased so it should be current. Thanks!

**Comment by benjaminws** [comment]
Getting to the point where this could be useful for our workflow. Thoughts on merging? I realize there's a new CLI on the horizon, but this tool provides a nice stop-gap.

**Comment by robzienert** [comment]
Forgot to merge. My b.

**Pull request merged**